samples/HelloAndroid/HelloAndroid.CommonTarget.sharpmake.cs

```diff
@@ -78,7 +81,7 @@ public override string Name
                     nameParts.Add(BuildSystem.ToString());
                 }
 
-                return string.Join(" ", nameParts);
+                return string.Join("_", nameParts);
```

Comment:
are spaces not handled well?

Reply:
It looks like we'll need to use underscore to join the configuration name since gradle is having trouble identify the space. I'm getting errors like 

> FAILURE: Build failed with an exception.
4>
4>    * What went wrong:
4>    Task 'assembleARM8Debug' is ambiguous in project ':app'. Candidates are: 'assembleArm8Debug_NoBlob_arm64_v8a', 'assembleArm8Debug_NoBlob_arm64_v8aDebug', 'assembleArm8Debug_NoBlob_arm64_v8aDebugAndroidTest', 'assembleArm8Debug_NoBlob_arm64_v8aDebugUnitTest', 'assembleArm8Debug_NoBlob_arm64_v8aRelease', 'assembleArm8Debug_NoBlob_arm64_v8aReleaseUnitTest'.

I'd imagine the task is like **Task 'assembleARM8Debug NoBlob_arm64_v8aDebug'** so and it's unable to differentiate from other task like **Task 'assembleARM8Debug NoBlob_arm64_v8aRelease'**. 
That's why I use underscore to join the configuration name, so that it'll become **Task 'assembleARM8Debug_NoBlob_arm64_v8aDebug'**, and gradle can differentiate it from other tasks

Reply:
But that's exactly why I put the spaces to make the config name, to make sure that all the location that need escaping are done so properly, maybe the generator needs some fixes

Reply:
There's nothing wrong with the spaces, it does the job as you mentioned. However, in this specific case **gradle** is the one that's having trouble parsing it. Not sure if it's a bug or something, if we use space here, it's not able to differentiate the task **'assembleARM8Debug NoBlob_arm64_v8aDebug'** from **'assembleARM8Debug NoBlob_arm64_v8aRelease'**. From **gradle**'s point of view, both of them are equal to **assembleARM8Debug**, everything after the space is ignored. That's why I change it to use underscore so that **gradle** can differentiate the tasks. Hope that makes sense

Reply:
Alright thanks! Please add this as a comment to explain why underscores are needed :)